src/LiquidityPool.sol

```diff
@@ -258,9 +279,10 @@ contract LiquidityPool is Auth, IERC7540 {
     }
 
     /// @inheritdoc IERC7575
-    function deposit(uint256 assets, address receiver) external returns (uint256 shares) {
-        shares = manager.deposit(address(this), assets, receiver, msg.sender);
-        emit Deposit(msg.sender, receiver, assets, shares);
+    function deposit(uint256 assets, address receiver, address owner) external returns (uint256 shares) {
```

Comment:
We need to also keep the `deposit(assets, receiver)` method, and add an overload for the one with `owner`